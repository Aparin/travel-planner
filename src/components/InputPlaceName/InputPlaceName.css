input.input {
    border: 2px lightblue solid;
    padding: 5px;
    width: 316px;
    display: block;
    margin: 0px auto;
    margin-bottom: 6px;
    height: 24px;
    font-family: Georgia, 'Times New Roman', Times, serif;
    font-size: 15px;
}