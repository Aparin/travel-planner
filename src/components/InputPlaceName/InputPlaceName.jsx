import React, { useState } from 'react';
import './InputPlaceName.css';
import PropTypes from 'prop-types';
import ErrorBoundary from '../ErrorBoundary';

const InputPlaceName = ({ addPoint }) => {
  const [state, setState] = useState({ value: '' });

  const nameToState = (e) => {
    if (e.keyCode !== 13 || e.target.value === '') return;

    addPoint(state.value);
    setState({ value: '' });
  };

  const onChange = (e) => {
    setState({
      value: e.target.value,
    });
  };

  return (
    <ErrorBoundary>
      <input
        type="text"
        className="input"
        id="NewPlaceName"
        placeholder={"Введите название места и нажмите 'Enter'"}
        onChange={onChange}
        onKeyUp={nameToState}
        value={state.value}
      />
    </ErrorBoundary>
  );
};

InputPlaceName.propTypes = {
  addPoint: PropTypes.func.isRequired,
};

export default InputPlaceName;
