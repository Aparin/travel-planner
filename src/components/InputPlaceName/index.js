import InputPlaceName from './InputPlaceName';

export default InputPlaceName;
