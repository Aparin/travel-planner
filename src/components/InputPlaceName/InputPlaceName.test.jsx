/* eslint-disable no-undef */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import InputPlaceName from './InputPlaceName';

describe('InputPlaceName', () => {
  it('renders InputPlaceName without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<InputPlaceName addPoint={() => {}} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('InputPlaceName snapshot', () => {
    const component = renderer.create(
      <InputPlaceName addPoint={() => {}} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
