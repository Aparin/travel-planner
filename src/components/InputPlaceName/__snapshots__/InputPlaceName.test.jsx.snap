// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputPlaceName InputPlaceName snapshot 1`] = `
<input
  className="input"
  id="NewPlaceName"
  onChange={[Function]}
  onKeyUp={[Function]}
  placeholder="Введите название места и нажмите 'Enter'"
  type="text"
  value=""
/>
`;
